### PR TITLE
Added address_record for bike

### PR DESCRIPTION
**Pull request by sethherr**
Address record was created and added for Users in https://github.com/bikeindex/bike_index/pull/2780

This adds address record to Bikes

- [x] Update bike create to use address record (and have BParam handle it)
- [x] Update bike update to use address record, add test to verify it doesn't create duplicate address records

#2922 is the follow up issue to track cleaning up things after the backfill and migration

### CODE REVIEW

File: app/views/bikes_edit/bike_details.html.haml [hunk 1]
```diff
@@ -168,34 +168,11 @@
...
-                  = f.text_field :city, class: "form-control", required: true, placeholder: t(".city")
-
-              .form-group.row
-                = f.label :zipcode, class: "form-well-label"
-                .form-well-input
-                  = f.text_field :zipcode, class: "form-control", required: true, placeholder: t(".zipcode")
+            -# It's sus to make queries in a page - but I think it's more important to call this at the correct place
+            - @bike.find_or_build_address_record(country_id: current_country_id)
+            = f.fields_for :address_record do |address_form|
+              = render(LegacyFormWell::AddressRecord::Component.new(form_builder: address_form, no_street: !BikeServices::Displayer.edit_street_address?(@bike, current_user)))
```
Comment (sethherr) [inline app/views/bikes_edit/bike_details.html.haml:174]: TODO: Verify that you don't want to pass in an organization here.
